fv3core/stencils/remapping_part2.py

```diff
@@ -27,125 +27,176 @@ def init_phis(hs: FloatField, delz: FloatField, phis: FloatField, te_2d: FloatFi
             phis = phis[0, 0, 1] - constants.GRAV * delz
 
 
-@gtstencil
 def sum_te(te: FloatField, te0_2d: FloatField):
     with computation(FORWARD):
         with interval(0, None):
             te0_2d = te0_2d[0, 0, -1] + te
 
 
-def compute(
-    qvapor: FloatField,
-    qliquid: FloatField,
-    qice: FloatField,
-    qrain: FloatField,
-    qsnow: FloatField,
-    qgraupel: FloatField,
-    qcld: FloatField,
-    pt: FloatField,
-    delp: FloatField,
-    delz: FloatField,
-    peln: FloatField,
-    u: FloatField,
-    v: FloatField,
-    w: FloatField,
-    ua: FloatField,
-    cappa: FloatField,
-    q_con: FloatField,
-    gz: FloatField,
-    pkz: FloatField,
-    pk: FloatField,
-    pe: FloatField,
-    hs: FloatFieldIJ,
-    te0_2d: FloatFieldIJ,
-    te: FloatField,
-    cvm: FloatField,
-    pfull: FloatFieldK,
-    ptop: float,
-    akap: float,
-    r_vir: float,
-    last_step: bool,
-    pdt: float,
-    mdt: float,
-    consv: float,
-    do_adiabatic_init: bool,
-):
-    saturation_adjustment = utils.cached_stencil_class(SatAdjust3d)(
-        cache_key="satadjust3d"
-    )
-
-    grid = spec.grid
-    copy_from_below(
-        ua, pe, origin=grid.compute_origin(), domain=grid.domain_shape_compute()
-    )
-    dtmp = 0.0
-    phis = utils.make_storage_from_shape(
-        pt.shape, grid.compute_origin(), cache_key="remapping_part2_phis"
-    )
-    te_2d = utils.make_storage_from_shape(
-        pt.shape[0:2], grid.compute_origin(), cache_key="remapping_part2_te_2d"
-    )
-    zsum1 = utils.make_storage_from_shape(
-        pt.shape[0:2], grid.compute_origin(), cache_key="remapping_part2_zsum1"
-    )
-    if spec.namelist.do_sat_adj:
-        fast_mp_consv = not do_adiabatic_init and consv > constants.CONSV_MIN
-        # TODO pfull is a 1d var
-        kmp = grid.npz - 1
-        for k in range(pfull.shape[0]):
-            if pfull[k] > 10.0e2:
-                kmp = k
-                break
-    if last_step and not do_adiabatic_init:
-        if consv > constants.CONSV_MIN:
-            raise NotImplementedError(
-                "We do not support consv_te > 0.001 "
-                "because that would trigger an allReduce"
-            )
-        elif consv < -constants.CONSV_MIN:
-            raise Exception(
-                "Unimplemented/untested case consv("
-                + str(consv)
-                + ")  < -CONSV_MIN("
-                + str(-constants.CONSV_MIN)
-                + ")"
-            )
+class Remapping_Part2:
+    def __init__(self):
+        self.grid = spec.grid
+        self.namelist = spec.namelist
```

Comment:
our convention has been lately to pass these in the init method. I know it's a little strange our convention for the grid has been to not pass it. So it is fine to do this here, we'll be doing a future pass to remove the global states and it doesn't hurt to do it this way, just wanted to mention it